tests/FieldAttributesValidationTest.php

```diff
@@ -0,0 +1,114 @@
+    {
+        $field = Field::text('name');
+
+        $this->assertNotSame(['required'], $field->getValidationRules());
+    }
+
+    /** @test */
+    function it_validate_the_existence_of_the_nullable_rule()
+    {
+        $field = Field::text('name');
+
+        $this->assertSame(['nullable'], $field->getValidationRules());
+    }
+
+    /** @test */
+    function it_validate_the_existence_of_the_email_rule()
+    {
+        $field = Field::email('email', ['required' => true]);
+
+        $this->assertEquals(['required', 'email'], $field->getValidationRules());
+    }
+
+    /** @test */
+    function it_validate_that_the_email_rule_does_not_exist()
+    {
+        $field = Field::text('name', ['required' => false]);
+
+        $this->assertNotEquals(['email'], $field->getValidationRules());
+    }
+
+    /** @test */
+    function it_validate_the_existence_of_the_url_rule()
+    {
+        $field = Field::url('link', ['required' => true]);
+
+        $this->assertEquals(['required', 'url'], $field->getValidationRules());
+    }
+
+    /** @test */
+    function it_validate_that_the_url_rule_does_not_exist()
+    {
+        $field = Field::text('name', ['required' => true]);
+
+        $this->assertNotEquals(['required', 'url'], $field->getValidationRules());
+    }
+
+    /** @test */
+    function it_there_are_multiples_rules()
+    {
+        $field = Field::email('email', ['required']);
+
+        $this->assertEquals(['required', 'email'], $field->getValidationRules());
+    }
+
+    /** @test */
+    function it_select_field_has_the_required_rule_and_options()
+    {
+        $field = Field::select('visibility', null, ['required'])->options([
+            'public' => 'Everyone',
+            'admin' => 'Admin only',
+            'auth' => 'Authenticated users only',
+            'guest' => 'Guest users only',
+        ]);
+
+        $this->assertContains('required', $field->getValidationRules());
+
+        $this->assertSame('in:"public","admin","auth","guest"', (string) $field->getValidationRules()[1]);
+    }
+
+    /** @test */
+    function it_select_field_no_have_rule_if_not_specific_option()
+    {
+        $field = Field::select('visibility')->options([
+            'public' => 'Everyone',
+            'admin' => 'Admin only',
+            'auth' => 'Authenticated users only',
+            'guest' => 'Guest users only',
+        ]);
+
+        $this->assertNotSame('in:"public","admin","auth","guest","moderator"', (string) $field->getValidationRules()[0]);
+    }
+
+    /** @test */
+    function it_select_field_have_rule_exists()
+    {
+        $field = Field::select('parent_id')
+            ->from('menu_items', 'label', 'id', function ($query) {
+                $query->whereNull('parent_id')
+                    ->orderBy('label', 'ASC');
+            })
+            ->label('Parent');
+
+        $this->assertInstanceOf(Exists::class, $field->getValidationRules()[1]);
```

Comment:
this requires at least 2 tests, one for a simple from, another one for a more complicated from. Test the string that is generated by the exists object. Please write more precise assertions.